src/core/application/hot/hot.ts

```diff
@@ -42,18 +52,26 @@ export class HotApplication {
         // Optionally refresh cache in background (fire and forget)
         this.refreshCacheInBackground(id, params);
 
+        // Fetch missing tagger users in background (fire and forget)
+        this.fetchUsersForTags(tags.slice(0, TOP_TAGS_TO_FETCH_USERS), params.user_id).catch((error) => {
+          Libs.Logger.warn('Background tagger fetch failed (non-critical)', { error });
+        });
+
         return tags;
       }
 
       // Cache miss - fetch from Nexus
       Libs.Logger.debug('Hot tags cache miss, fetching from Nexus', { id });
       const tags = await Core.NexusHotService.fetch(params);
 
-      // Store in cache (fire and forget)
       if (tags.length > 0) {
-        Core.LocalHotService.upsert(id, tags).catch((error) => {
-          Libs.Logger.error('Failed to cache hot tags', { id, error });
-        });
+        // Fetch and persist users first, then persist hot tags (fire and forget)
+        // This prevents race conditions where liveQuery triggers before users are cached
+        // Not using await to prevent blocking the UI, tags are returned immediately
+        this.fetchUsersForTags(tags.slice(0, TOP_TAGS_TO_FETCH_USERS), params.user_id)
+          .catch((error) => Libs.Logger.warn('Background tagger fetch failed', { error }))
+          .then(() => Core.LocalHotService.upsert(id, tags))
+          .catch((error) => Libs.Logger.error('Failed to cache hot tags', { id, error }));
       }
```

Comment:
Make this blocking just to be sure because we dont know what the forntend will do